### ISSUE TITLE
Closed caption casting

### ISSUE THREAD

**Issue by pantlalitm** [pranav-prakash/RokuCast#15]
This is a great extension. haven't found anything similar for chrome yet. Lots of websites support closed captioning on the videos, but they do not appear on casted videos. Is it a problem with Play on Roku protocol or can it be overcome by updating the extension. 

Also, why not submit this extension to chrome store. make it official :+1:

**Comment by pranav-prakash** [comment]
Can you link me a site which has captions on the video? It should definitely be possible to add it in assuming it's in a format that Roku can handle (ttml).

**Comment by pantlalitm** [comment]
For example, fmovies.to has closed captioning option on most of its videos. I don't know the format of subtitles in these videos.

There are some other website(can't remember right now) which allow you to upload a srt file, and get the captions. I think roku supports srt, so subtitles on these websites should work hopefully

**Comment by pranav-prakash** [comment]
Ok so for fmovies.to which uses jwplayer, by default the caption is in WebVTT format which roku does not support for mp4.

BUT, clicking the download caption button inside the player gives you access to an srt formatted file, (whose general url is like: http://fmovies.to/subtitle/[ID].dl) which can then be sent to the roku.

I thus can add in support specifically for fmovies.to since they seem to have a nice way to get the srt files directly. 

In general however, it may be hard to implement a universal method for doing this because most sites seem to use webvtt.

**Comment by pantlalitm** [comment]
Instead of making it specific to fmovies.to, I think it would be beneficial to add general support to cast srt files with a webvideo in Rokucast. If that can be added to RokuCast, one can Download srt files to PC and send it to roku with a web video, and/or, one can also send a web link to the srt file.

**Comment by pranav-prakash** [comment]
>Cast a web link that you supply to the srt file along with the video

That would be a good idea. Also relatively easy to implement.

The only downside is that it's not automated — you would have to search and supply a link to a srt file yourself. Moreover, it would have to be a link accessible directly (i.e. not downloaded through a file hosting provider or something). I'm not sure how easy it is to find those online, so the scope and usability could be limited.

The other option you mentioned of sending SRT files downloaded to the PC might not be feasible within the scope chrome extension. You can only really send URLs of things to the roku, not the files themselves. If you were to go this route you would probably have to create a local webserver from which the file could be accessed on the Roku — not really feasible within the sandbox of an extension.

**Comment by dhruvkp** [comment]
Could you please tell me more about how to send caption stream to roku? 
Like I know we can send video URL like this- ipaddress:8060/input/15985?t=v&u="url"
What url parameters can be used. to send caption stream? Thanks!

**Comment by pranav-prakash** [comment]
I don't believe the inbuilt play on roku channel endpoint (the one which responds to a video url sent to "ipaddress:8060/input/15985...") has a parameter for captions. However, it should not be too hard to modify the custom receiver channel (found on the /master branch of this repo) to support this (it would just amount to passing the url as an extra parameter which is parsed in [main.brs](https://github.com/pranav-prakash/RokuCast/blob/master/RokuCast%20Receiver/source/main.brs) and accordingly setting the video metadata to include the captions. As noted before the caption formats roku supports can be found [here](https://sdkdocs.roku.com/display/sdkdoc/Closed+Caption+Support]).
